#162948833 Heroku deployment

d-kahara opened the pull request:
**What does this PR do?**
Setup heroku app configurations for staging and production environments

**Description of Tasks to be completed?**

- Install development dependencies i.e django-heroku(automatically configures DATABASE_URL, 
  ALLOWED_HOSTS, WhiteNoise (for static assets), Logging, and Heroku CI for the application)
- Create a Procfile with necessary settings
- Setup pipelines for staging and production environments
- Configure postgres addon for heroku
- Integrate heroku with github repo 

#### How should this be manually tested?

- Navigate to this url: https://ah-django-unchained-staging.herokuapp.com/

**What are the relevant pivotal tracker stories**

[##162948833](https://www.pivotaltracker.com/story/show/162948833)

**Screenshots**
![image](https://user-images.githubusercontent.com/27801956/50927177-0e43f600-1468-11e9-9b18-423bfb352991.png)

kwameasiago commented:
Use git commit --no-edit instead of having two commits that have the same message.

andrewhingah reviewed (approved):
Well written commit message. Just consider using `git commit --no edit` to avoid having two commits with similar message.

kwameasiago reviewed (approved):
Looks good to me

MaggieChege reviewed (approved):
Look good to me as well.

ignatiusukwuoma commented on authors/apps/authentication/serializers.py:
```diff
@@ -34,6 +34,7 @@ class LoginSerializer(serializers.Serializer):
     email = serializers.CharField(max_length=255)
     username = serializers.CharField(max_length=255, read_only=True)
     password = serializers.CharField(max_length=128, write_only=True)
+    
```
Why a blank line addition here?

ignatiusukwuoma replied:
Remove unnecessary white-spaces

ignatiusukwuoma replied:
Please remove your edit on this file. Your task doesn't require you to edit this file

ignatiusukwuoma commented on authors/settings.py:
```diff
@@ -146,3 +148,5 @@
     # ),
 }
+django_heroku.settings(locals())
+
```
Why add a blank line after every piece of code?

d-kahara replied:
Error on my part, I had misread the PEP 8 guidelines on [blank](https://www.python.org/dev/peps/pep-0008/#blank-lines)  lines.

ignatiusukwuoma commented:
Rebase this branch on master carefully and ensure that no package in the requirements.txt is deleted

ignatiusukwuoma commented on requirements.txt:
```diff
@@ -5,4 +5,10 @@ djangorestframework==3.9.0
+pytz==2018.9
+whitenoise==4.1.2
+gunicorn==19.9.0
+django-heroku==0.3.1
```
We have django-heroku twice in this file

d-kahara replied:
> Are we supposed to have DEBUG=True in production?

We'll change this when we're ready to ship for production?

ignatiusukwuoma commented:
Are we supposed to have DEBUG=True in production?

ignatiusukwuoma commented:
Your Travis build should pass now since all tests are passing

d-kahara committed:
chore(devops):Configure heroku hosting settings

 -Install djengo-heroku dependency
 -Create Procfile and configure dynos
 -Install decpoule dependency
 -Create Heroku app and Configure postgres addon for heroku

[Delivers #162948833]

ignatiusukwuoma commented:
Let me know when this PR is ready to be merged

d-kahara commented:
> Let me know when this PR is ready to be merged

Ready to merge
